src/command.spec.ts

```diff
@@ -49,179 +52,125 @@ describe('#start()', () => {
 
     it('sets stdin, process and PID', () => {
         const command = createCommand();
-
         command.start();
+
         expect(command.process).toBe(process);
         expect(command.pid).toBe(process.pid);
         expect(command.stdin).toBe(process.stdin);
     });
 
-    it('shares errors to the error stream', done => {
+    it('shares errors to the error stream', () => {
         const command = createCommand();
-        command.error.subscribe(data => {
-            expect(data).toBe('foo');
-            expect(command.process).toBeUndefined();
-            done();
-        });
-
+        const observerSpy = subscribeSpyTo(command.error);
         command.start();
         process.emit('error', 'foo');
+
+        expect(observerSpy.getFirstValue()).toBe('foo');
+        expect(command.process).toBeUndefined();
     });
 
-    it('shares start and close timing events to the timing stream', done => {
+    it('shares start and close timing events to the timing stream', () => {
         const command = createCommand();
-
+        const observerSpy = subscribeSpyTo(command.timer);
         const startDate = new Date();
         const endDate = new Date(startDate.getTime() + 1000);
         jest.spyOn(Date, 'now')
             .mockReturnValueOnce(startDate.getTime())
             .mockReturnValueOnce(endDate.getTime());
-
-        let callCount = 0;
-        command.timer.subscribe(({ startDate: actualStartDate, endDate: actualEndDate }) => {
-            switch (callCount) {
-                case 0:
-                    expect(actualStartDate).toStrictEqual(startDate);
-                    expect(actualEndDate).toBeUndefined();
-                    break;
-                case 1:
-                    expect(actualStartDate).toStrictEqual(startDate);
-                    expect(actualEndDate).toStrictEqual(endDate);
-                    done();
-                    break;
-                default:
-                    throw new Error('Unexpected call count');
-            }
-            callCount++;
-        });
-
         command.start();
         process.emit('close', 0, null);
+
+        expect(observerSpy.getValueAt(0)).toEqual({ startDate, endDate: undefined });
+        expect(observerSpy.getValueAt(1)).toEqual({ startDate, endDate });
```

Comment:
This is nice! But it seems it assumes that the observers will emit synchronously... or did I miss anything?

Reply:
Oh, you're right! Good catch, thanks! I'm going to address this in the next days...

Reply:
So what do you think about this? => #369 😃